### PR TITLE
[stable] Remove superfluous core.stdc.* dependencies in std.{conv,random}

### PR DESCRIPTION
**Pull request by kinke**
For WebAssembly etc.

### CODE REVIEW

File: std/conv.d [hunk 1]
```diff
@@ -3240,7 +3239,7 @@ if (isInputRange!Source && isSomeChar!(ElementType!Source) && !is(Source == enum
         }
     }
   L6: // if overflow occurred
-    enforce(ldval != HUGE_VAL, new ConvException("Range error"));
+    enforce(ldval != real.infinity, new ConvException("Range error"));
```
Comment (kinke) [inline std/conv.d]: `HUGE_VAL` is `double.infinity`, `ldval` is a `real`.

File: std/random.d [hunk 2]
```diff
@@ -2182,8 +2182,7 @@ do
         immutable T u = (rng.front - rng.min) * factor;
         rng.popFront();
 
-        import core.stdc.limits : CHAR_BIT;  // CHAR_BIT is always 8
```
Comment (atilaneves) [inline std/random.d]: What if on some platform, someday, it isn't 8?

Comment (kinke) [reply std/random.d]: Then `T.sizeof` wouldn't be in 8-bit bytes anymore, and all hell breaks loose I guess.

**Comment by wilzbach** [comment]
(The DAutoTest failure is unrelated and can be ignored.)

**Pull request merged**